src/pybind/rados/rados.pyx

```diff
@@ -2803,6 +2823,44 @@ returned %d, but should return zero on success." % (self.name, ret))
         else:
             raise LogicError("Ioctx.write_full(%s): rados_write_full \
 returned %d, but should return zero on success." % (self.name, ret))
+    
+    @requires(('key', str_type), ('data', bytes), ('write_len', int), ('offset', int))
+    def writesame(self, key, data, write_len, offset=0):
```

Comment:
appreciate your contribution, but can you explain why there are two definitions of `writesame` or should it be `writesame_op` instead as used in tests?

Reply:
@ideepika
1. writesame(self, key, data, write_len, offset=0) is method of class Ioctx, writesame2 is method of class WriteOp , so there is no conflict. The relationship beween Ioctx::writesame() and WriteOp::writesame() is equal to that betwen Ioctx::write_full() and WriteOp::write_full().

2. test_writesame() is the test function of Ioctx::writesame()

```
    def test_writesame(self):
        self.ioctx.writesame('ob', b'rzx', 9)
        eq(self.ioctx.read('ob'), b'rzxrzxrzx')

```
3. test_writesame_op() is the test function of WriteOp::writesame()

```
    def test_writesame_op(self):
        with WriteOpCtx() as write_op:
            write_op.writesame(b'rzx', 9)
            self.ioctx.operate_write_op(write_op, 'abc')
            eq(self.ioctx.read('abc'), b'rzxrzxrzx')
```

Reply:
1. writesame(self, key, data, write_len, offset=0) is method of class Ioctx, writesame(self, to_write, write_len, offset=0) is method of class WriteOp , so there is no conflict. The relationship beween Ioctx::writesame() and WriteOp::writesame() is equal to that betwen Ioctx::write_full() and WriteOp::write_full().

2. test_writesame() is the test function of Ioctx::writesame()
  ```
  def test_writesame(self):
        self.ioctx.writesame('ob', b'rzx', 9)
        eq(self.ioctx.read('ob'), b'rzxrzxrzx')
```

3. test_writesame_op() is the test function of WriteOp::writesame()
   ```
 def test_writesame_op(self):
        with WriteOpCtx() as write_op:
            write_op.writesame(b'rzx', 9)
            self.ioctx.operate_write_op(write_op, 'abc')
            eq(self.ioctx.read('abc'), b'rzxrzxrzx')
```

Reply:
1. writesame(self, key, data, write_len, offset=0) is method of class Ioctx, writesame(self, to_write, write_len, offset=0) is method of class WriteOp , so there is no conflict. The relationship beween Ioctx::writesame() and WriteOp::writesame() is equal to that betwen Ioctx::write_full() and WriteOp::write_full().
2. test_writesame() is the test function of Ioctx::writesame()
```
def test_writesame(self):
      self.ioctx.writesame('ob', b'rzx', 9)
      eq(self.ioctx.read('ob'), b'rzxrzxrzx')
```
3. test_writesame_op() is the test function of WriteOp::writesame() 
```
def test_writesame_op(self):
      with WriteOpCtx() as write_op:
             write_op.writesame(b'rzx', 9)
             self.ioctx.operate_write_op(write_op, 'abc')
             eq(self.ioctx.read('abc'), b'rzxrzxrzx')
```

Reply:
@ideepika Actually, I don't understand why you asked me to squash WriteOp::writesame() and Ioctx::writesame() into single commit.

Reply:
@ideepika   Thanks for your suggestions.

Reply:
> @ideepika Actually, I don't understand why you asked me to squash WriteOp::writesame() and Ioctx::writesame() into single commit.

thanks for the explanation, I misunderstood your code, my mistake. Yeah it would be relevant to them differently